### ISSUE TITLE
Automate the publication of binary artifacts from CI

### ISSUE THREAD

**Issue by labra** [weso/utils#21]
The publication of artifacts could be automated from the continuous integration process to avoid manual errors. 

One important aspect is to remove coverage code before publishing by activating `coverageOff`.

**Comment by thewilly** [comment]
If we use automatic deployment for artifact publication we must add some kind of safe guard against no-test-passing code on Master. Else if non working code gets there can make fail any other project relying on this.

**Comment by labra** [comment]
Yes, I think we should publish only when all the tests pass and the version in file [version.sbt](https://github.com/weso/utils/blob/master/version.sbt) changes.

Maybe [sbt-dynver](https://github.com/dwijnand/sbt-dynver) can help to keep the version updated, but I am not sure yet.

**Comment by labra** [comment]
I will try to automate this using [sbt-github-actions](https://github.com/djspiewak/sbt-github-actions), [sbt-github-packages](https://github.com/djspiewak/sbt-github-packages) and [sbt-ci-release](https://github.com/olafurpg/sbt-ci-release)
